astropy/table/index.py

```diff
@@ -182,7 +182,7 @@ def insert_row(self, pos, vals, columns):
         key = [None] * len(self.columns)
         for i, col in enumerate(columns):
             try:
-                key[i] = vals[self.col_position(col.info.name)]
```

Comment:
The `i` variable is with respect to the original table `columns` and input `vals` (tuple of items to be inserted), while `key` has the length of the number of index columns.  `self.col_position(name)` gives the index position with respect to the list of *index columns* for the column `name`.